internal/controller/commitstatus_controller.go

```diff
@@ -65,36 +66,56 @@ func (r *CommitStatusReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	err := r.Get(ctx, req.NamespacedName, &cs, &client.GetOptions{})
 	if err != nil {
 		if errors.IsNotFound(err) {
-			logger.Info("CommitStatus not found", "namespace", req.Namespace, "name", req.Name)
+			logger.Info("CommitStatus not found")
 			return ctrl.Result{}, nil
 		}
 
-		logger.Error(err, "failed to get CommitStatus", "namespace", req.Namespace, "name", req.Name)
+		logger.Error(err, "failed to get CommitStatus")
 		return ctrl.Result{}, err
 	}
 
 	// We use observed generation pattern here to avoid provider API calls.
 	if cs.Status.ObservedGeneration == cs.Generation {
-		logger.Info("No need to reconcile", "namespace", req.Namespace, "name", req.Name)
+		logger.Info("No need to reconcile")
 		return ctrl.Result{}, nil
 	}
 
-	commitStatusProvider, err := r.getCommitStatusProvider(ctx, cs)
-	if err != nil {
-		return ctrl.Result{}, err
+	if cs.Spec.Sha == "" || cs.Spec.Phase == "" {
+		logger.Info("Skip setting commit status, missing sha or phase")
+		return ctrl.Result{}, nil
```

Comment:
Do we already have apply-time validation for these?

Reply:
So phase defaults to pending which is find so it should not ever be an empty string because it is also an enum.

Sha was a bit trickier because it is required but defaults to empty string idk enough about what end users would do when creating a commits status. I actually at one point figured out to regex it to a sha but then I figured if people might not know the shaw at creation. So I just went with this.

Reply:
No worries, honestly a comment saying `empty phase should be impossible, checking anyway` might be fine.